ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,228 @@
+
+	mockComparer := testTaskManifestResponder.taskComparer.(*mock_session.MockTaskComparer)
+	mockComparer.EXPECT().CompareRunningTasksOnInstanceWithManifest(gomock.Any()).Times(1)
+
+	// Handle the task manifest message update
+	handleManifestMessage := testTaskManifestResponder.HandlerFunc().(func(*ecsacs.TaskManifestMessage))
+	go handleManifestMessage(testManifest)
+
+	// Wait for the ack to send through our channel so we can receive it
+	manifestMessageAckSent := <-ackSent
+	require.Equal(t, aws.StringValue(manifestMessageAckSent.MessageId), aws.StringValue(testManifest.MessageId))
+
+}
+
+// TestTaskManifestSequenceNumbers will make sure the manifest can update, retrieve and catch stale
+// Manifest sequence numbers appropriately.
+func TestTaskManifestSequenceNumbers(t *testing.T) {
+
+	// Create a waitgroup to make sure the sequence number has updated
+	// before we try to access it
+	var ackWg sync.WaitGroup
+	ackWg.Add(1)
+
+	testResponseSender := func(response interface{}) error {
+		defer ackWg.Done()
+		return nil
+	}
+
+	testTaskManifestResponder := setupTestTaskManifestResponder(t, metrics.NewNopEntryFactory(), testResponseSender)
+
+	mockmmIDA := testTaskManifestResponder.messageIDAccessor.(*mock_session.MockManifestMessageIDAccessor)
+	mockmmIDA.EXPECT().SetMessageID(gomock.Any()).Times(1)
+
+	// Expect GetLatestSequenceNumber once in handleManifestMessage, and once
+	// when we validate it
+	mockSNA := testTaskManifestResponder.snAccessor.(*mock_session.MockSequenceNumberAccessor)
+	mockSNA.EXPECT().GetLatestSequenceNumber().Times(2)
+	mockSNA.EXPECT().SetLatestSequenceNumber(aws.Int64(20)).Times(1)
+
+	mockComparer := testTaskManifestResponder.taskComparer.(*mock_session.MockTaskComparer)
+	mockComparer.EXPECT().CompareRunningTasksOnInstanceWithManifest(gomock.Any()).Times(1)
+
+	// Create test task manifest.
+	testManifest := &ecsacs.TaskManifestMessage{
+		ClusterArn:           aws.String(testconst.ClusterName),
+		ContainerInstanceArn: aws.String(testconst.ContainerInstanceARN),
+		MessageId:            aws.String(testconst.MessageID),
+		Tasks: []*ecsacs.TaskIdentifier{
+			{DesiredStatus: aws.String("STOPPED"), TaskArn: aws.String(testconst.TaskARN)},
+		},
+		Timeline: aws.Int64(20),
+	}
+
+	// Handle the task manifest message update
+	handleManifestMessage := testTaskManifestResponder.HandlerFunc().(func(*ecsacs.TaskManifestMessage))
+	handleManifestMessage(testManifest)
+
+	ackWg.Wait()
+
+	// Now assert that the latest sequence number has been updated
+
+	// Now manually try to reset the sequence and make sure it does not update
+	// As 15 < 20 we should not be able to update to an older sequence number
+	testTaskManifestResponder.snAccessor.SetLatestSequenceNumber(testManifest)
+
+	require.NotEqual(t,
+		aws.Int64(testTaskManifestResponder.snAccessor.GetLatestSequenceNumber()),
+		aws.Int64(StartingLatestTaskManifestSequenceNumber))
+}
+
+func TestTaskStopVerification(t *testing.T) {
```

Comment:
Could we combine this test with TestManifestAckHappyPath into a single happy path test? A happy path of handling a task manifest message involves both ACKing the task manifest message as well as sending a task stop verification message (if there is more than 1 task to stop) and I feel there would be less duplication/lines of code as well.